### ISSUE TITLE
Add Valkey support to ElastiCache `CacheCluster`

### ISSUE THREAD

**Issue by flostadler** [pulumi/pulumi-aws#4771]
Currently the ElastiCache `CacheCluster` resource does not support the Valkey engine. Instead it only supports redis and memcached.
Support for this is currently blocked on AWS APIs not accepting the engine type `valkey` in the `CreateCacheCluster` API call.

This upstream issue (https://github.com/hashicorp/terraform-provider-aws/issues/39905) tracks support for this. Once it's released in the upstream terraform provider we can release it as part of `pulumi-aws`.

As a workaround you can use the ElastiCache `ReplicationGroup` resource to deploy Valkey cache cluster today. By setting `numCacheClusters` to one, it creates a single node cluster.

```ts
import * as aws from "@pulumi/aws";

const valkey = new aws.elasticache.ReplicationGroup("valkey-test", {
    description: "valkey cluster",
    nodeType: "cache.m4.large",
    numCacheClusters: 1,
    engine: "valkey",
    port: 6379,
});
```

**Comment by mikstew** [comment]
@flostadler It looks like a fix for https://github.com/hashicorp/terraform-provider-aws/issues/39905 has been released. Any timeline for an update to add Valkey support? Thanks!

**Comment by arjun810** [comment]
It'd be nice if CacheCluster supported valkey. Anything we can to do help with that?

**Comment by corymhall** [comment]
Reading through that issue it looks like it should now be supported. Are you able to try it out and let me know if you run into any errors?

**Comment by arjun810** [comment]
still getting this:

creating ElastiCache Cache Cluster (<>): operation error ElastiCache: CreateCacheCluster, https response error StatusCode: 400, RequestID: <>, InvalidParameterValue: This API doesn't support Valkey engine. Please use CreateReplicationGroup API for Valkey cluster creation.

**Comment by guineveresaenger** [comment]
I've just confirmed that the following program still errors:

```typescript
import * as aws from "@pulumi/aws"

const valkeyCluster = new aws.elasticache.Cluster("valkey-test", {
    engine: "valkey",
    engineVersion: "7.2",
    nodeType: "cache.t3.micro",
    numCacheNodes: 1,
    port: 6379,
});

export const clusterId = valkeyCluster.clusterId;
export const cacheNodes = valkeyCluster.cacheNodes;
```

Output, where they explicity state to use `CreateReplicationGroup API`.

```sh

     Type                        Name          Status                  Info
 +   pulumi:pulumi:Stack         aws-4771-dev  **creating failed**     1 error
 +   └─ aws:elasticache:Cluster  valkey-test   **creating failed**     2 errors

Diagnostics:
  pulumi:pulumi:Stack (aws-4771-dev):
    error: update failed

  aws:elasticache:Cluster (valkey-test):
    error:   sdk-v2/provider2.go:572: sdk.helper_schema: creating ElastiCache Cache Cluster (valkey-test-74863c3): operation error ElastiCache: CreateCacheCluster, https response error StatusCode: 400, RequestID: da23787b-b8df-4f1a-adbb-fcf8fc77d8b1, InvalidParameterValue: This API doesn't support Valkey engine. Please use CreateReplicationGroup API for Valkey cluster creation.: provider=aws@7.28.0
    error: 1 error occurred:
    	* creating ElastiCache Cache Cluster (valkey-test-74863c3): operation error ElastiCache: CreateCacheCluster, https response error StatusCode: 400, RequestID: da23787b-b8df-4f1a-adbb-fcf8fc77d8b1, InvalidParameterValue: This API doesn't support Valkey engine. Please use CreateReplicationGroup API for Valkey cluster creation.

    [Pulumi Neo] Would you like help with these diagnostics?
    https://app.pulumi.com/guinevere-pulumi-corp/aws-4771/dev/updates/1?explainFailure

Resources:
    + 1 created
    2 errored

Duration: 5s
```

pulumi about

```
Dependencies:
NAME            VERSION
@pulumi/aws     7.28.0
@pulumi/pulumi  3.235.0
@types/node     18.19.130
typescript      5.9.3
```

It appears as though https://github.com/hashicorp/terraform-provider-aws/issues/39905 was closed prematurely, or with incomplete resolution.

We would encourage anyone to file a new issue upstream. 🙏 

Update: I pulled the [AWS docs](https://docs.aws.amazon.com/AmazonElastiCache/latest/APIReference/API_CreateCacheCluster.html) and there doesn't look to be native support on the AWS API either. This is blocked until implemented via AWS.
